pymaster/field.py

```diff
@@ -658,3 +685,279 @@ def get_ell_sampling(self):
         """
         ells = lib.get_ell_sampling_flat(self.fl, int(self.fl.fs.n_ell))
         return ells
+
+
+class NmtFieldCatalog(NmtField):
+    """ An :obj:`NmtFieldCatalog` object contains all the information
+    describing a given field sampled at the discrete positions of
+    a given catalog of sources. It can also be used for clustering studies,
+    in which the density of points itself is the field of interest.
+
+    Args:
+        positions (`array`): Source positions, provided as a list or array
+            of 2 arrays. If ``lonlat`` is True, the arrays should contain the
+            longitude and latitude of the sources, in this order, and in
+            degrees (e.g. R.A. and Dec. if using Equatorial coordinates).
+            Otherwise, the arrays should contain the colatitude and
+            longitude of each source in radians (i.e. the spherical
+            coordinates :math:`(\\theta,\\phi)`).
+        weights (`array`): An array containing the weight assigned to
+            each source.
+        field (`array`): A list of 1 or 2 arrays (for spin-0 and spin-s
+            fields, respectively), containing the value of the field
+            whose power spectra we aim to calculate at the positions of
+            each source.
+        lmax (:obj:`int`): Maximum multipole up to which the spherical
+            harmonics of this field will be computed.
+        lmax_mask (:obj:`int`): Maximum multipole up to which the spherical
+            harmonics of this field's mask will be computed. If negative,
+            it will default to ``4*lmax``.
```

Comment:
Change to `lmax`

Reply:
Ah right. Yes, here would be good. Let me do that.